usim/_primitives/activity.py

```diff
@@ -51,16 +51,37 @@ def __transcript__(self) -> ActivityCancelled:
 
 
 class ActivityExit(BaseException):
-    ...
+    """A :py:class:`~.Activity` forcefully exited"""
 
 
-class Activity(Condition, Generic[RT]):
+class Activity(Awaitable[RT]):
     """
-    Active coroutine that allows others to listen for its completion
+    Concurrently running activity that allows multiple activities to await its completion
 
-    :note: Simulation code should never instantiate this class directly.
+    A :py:class:`Activity` represents an activity that is concurrently run in a :py:class:`~.Scope`.
+    This allows to store or pass an an :py:class:`Activity`, in order to check its progress.
+    Other activities can ``await`` a :py:class:`Activity`,
+    which returns any results or exceptions on completion, similar to a regular activity.
+
+    .. code:: python3
+
+        await my_activity()  # await a bare activity
+
+        async with Scope() as scope:
+            activity = scope.do(my_activity())
+            await activity   # await a rich activity
```

Comment:
How is my_activity defined? Is this of type `Activity`?
What is a bare and a rich activity? It has not been explained before.